lib/ContentTypeParser.js

```diff
@@ -46,14 +46,16 @@ ContentTypeParser.prototype.getHandler = function (contentType) {
 }
 
 ContentTypeParser.prototype.run = function (contentType, handler, context, params, req, res, query) {
-  this.getHandler(contentType)(req, done)
-
-  function done (body) {
-    if (body instanceof Error) {
+  var result = this.getHandler(contentType)(req, done)
+  if (result && typeof result === 'function') {
```

Comment:
I would check `result.then`.

Reply:
My fault, was I typo in my suggestion above :D